apps/example/src/domain/invoice.ts

```diff
@@ -1,190 +1,207 @@
 import { PayerId, InvoiceId } from "./identifiers.js"
 import z from "zod"
-import { Codec, Decider, LoadOption, StreamId } from "@equinox-js/core"
+import { Codec, Decider, LoadOption, StreamId, StreamName } from "@equinox-js/core"
 import { reduce } from "ramda"
 import * as Config from "../config/equinox.js"
 
-export const CATEGORY = "Invoice"
-export const streamId = StreamId.gen(InvoiceId.toString)
-type InvoiceRaised = { payer_id: PayerId; amount: number }
-type Payment = { amount: number; reference: string }
-
-export type Event =
-  | { type: "InvoiceRaised"; data: InvoiceRaised }
-  | { type: "PaymentReceived"; data: Payment }
-  | { type: "InvoiceFinalized" }
-
-export const RaisedSchema = z.object({
-  payer_id: z.string().transform(PayerId.parse),
-  amount: z.number(),
-})
-export const PaymentSchema = z.object({ reference: z.string(), amount: z.number() })
-
-export const codec = Codec.zod<Event>({
-  InvoiceRaised: RaisedSchema.parse,
-  PaymentReceived: PaymentSchema.parse,
-  InvoiceFinalized: () => undefined,
-})
-
-// Fold
-export type InvoiceState = {
-  amount: number
-  payer_id: PayerId
-  payments: Set<string>
-  amount_paid: number
+export namespace Stream {
+  export const CATEGORY = "Invoice"
+  export const streamId = StreamId.gen(InvoiceId.toString)
+  export const parseId = StreamName.dec(CATEGORY, InvoiceId.parse)
```

Comment:
is there no 'try` idiom? 
is `parse` understood to be a process that can fail/reject?

Reply:
yeah, parse is understood as having the possibility to throw. E.g. the `z.string().uuid().parse('1')` would throw a `ZodError`

Reply:
ok but this is not throwing for malformed - it's seeing whether the category matches ? in ruby there would be a `?` in the name ;)

Reply:
> ok but this is not throwing for malformed

It is throwing on malformed, It's just not throwing if the category doesn't match

Reply:
what do you call a function that can throw and returns an option? (don't say "stupid")

Reply:
tryParse
tryMatch
parseIf